### DISCUSSION TITLE
URL link annotation not opening on MacOS preview app, but is on Google Chrome

### DISCUSSION THREAD

**Discussion by ghost** [Hopding/pdf-lib#1522] [Q&A]
The following link is not opening using MacOS preview app but is using Google Chrome. Am I doing something wrong, or is this an issue with the library?

const link = page.doc.context.register(
    firstPage.doc.context.obj({
      Type: 'Annot',
      Subtype: 'Link',
      Rect: [x, y, x + textWidth, y + textHeight],
      Border: [0, 0, 0],
      C: [0, 0, 1],  // Color in RGB; this is blue. You can adjust as needed.
      A: {
        Type: 'Action',
        S: 'URI',
        URI: 'https://example.com',
      },
    }),
  );
  firstPage.node.set(PDFName.of('Annots'), document.context.obj([link]));

**Comment by anish-ttt** [comment]
Same issue happening for me. On hovering in Mac Preview, the cursor changes identifying there is a link present, but nothing happens on clicking on it.

Works fine everywhere else.
